### PR TITLE
Add a 'get started' vignette

### PR DESCRIPTION
**Pull request by paul-buerkner**
As promised, this PR adds (an initial draft) of a 'Get Started' vignette. I hope this example captures the main parts of the intended workflow reasonably well.

### CODE REVIEW

**Comment by hsbadr** [comment]
Thanks! I think the check fails because this uses packages that we don't suggest/import:
```r
Error: processing vignette 'GetStarted.Rmd' failed with diagnostics:
there is no package called 'recipes'
```

File: vignettes/GetStarted.Rmd [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    keep_md: true
+vignette: >
+  %\VignetteIndexEntry{Getting Started}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r}
+library(recipes)
+library(workflows)
```
Comment (hsbadr) [inline vignettes/GetStarted.Rmd:18]: We need to add `recipes` and `workflows` to `Suggests`.

File: vignettes/GetStarted.Rmd [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+in the data. In a first step, we use the `recipes` package to prepare (a recipe
+for) the `epilepsy` data. This data set is shipped with the `brms` package,
+which is automatically loaded by `bayesian`.
+
+```{r}
+epi_recipe <- epilepsy %>% 
+  recipe() %>%
+  update_role(count, new_role = "outcome") %>%
+  update_role(Trt, Age, Base, patient, new_role = "predictor") %>%
+  update_role(patient) %>%
```
Comment (hsbadr) [inline vignettes/GetStarted.Rmd]: The default `role` is `predictor`; so, let's add explicit roles and may add a role as `group`.
```suggestion
  update_role(patient, new_role = "predictor") %>%
  add_role(patient, new_role = "group") %>%
```

**Comment by paul-buerkner** [comment]
Right. Should be fixed.

Am So., 16. Mai 2021 um 20:20 Uhr schrieb Hamada S. Badr <
***@***.***>:

File: vignettes/GetStarted.Rmd [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+Above, we not only define the roles of the relevant variables but also 
+normalized the `Age` and `Base` predictors to facilitate model fitting later
+on. In the next step, we use `bayesian` to set up a basic model structure.
+
+```{r}
+epi_model <- bayesian() %>%
+  set_engine("brms") %>%
+  set_mode("regression") %>%
+  update(family = poisson())
```
Comment (hsbadr) [inline vignettes/GetStarted.Rmd]: Why do you want to define the arguments by `update` not inside `bayesian`? This means that you specify the default arguments and then update them other than specifying the model at the first call. Both should be identical,

File: vignettes/GetStarted.Rmd (hunk 2 above)
Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: I tried role `group` and it failed later on when trying to fit the model. Does it work for you?

File: DESCRIPTION [hunk 4]
```diff
@@ -35,6 +35,8 @@ Imports:
     utils
 Suggests:
     covr,
+    recipes,
+    workflows,
```
Comment (hsbadr) [inline DESCRIPTION]: Let's keep this list in an alphabetical order.

File: vignettes/GetStarted.Rmd (hunk 3 above)
Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: True but I want to show users how they can later on update an existing model by overwriting only specific arguments. I am fine with either option of course.

File: DESCRIPTION (hunk 4 above)
Comment (paul-buerkner) [reply DESCRIPTION]: Done.

File: vignettes/GetStarted.Rmd (hunk 2 above)
Comment (hsbadr) [reply vignettes/GetStarted.Rmd]: You need to add a "group" role after setting it as a predictor (to make it available in the processed data passed to the model fit, `bayesian_fit`). Any new roles should be ok.

Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: I see. Let me try.

Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: Nope, still fails for me unfortunately.

Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: Ah wait. I need to `add` the role. Sorry.

File: vignettes/GetStarted.Rmd (hunk 3 above)
Comment (hsbadr) [reply vignettes/GetStarted.Rmd]: Sounds good. I like the idea of showing the case for `update` but the users might think it should be applied everytime they build a model. I think we could specify the model with `bayesian(family = poisson())` and show the users how to change/update the `family` in the following step.

File: vignettes/GetStarted.Rmd (hunk 2 above)
Comment (hsbadr) [reply vignettes/GetStarted.Rmd]: Yes, `update_role` sets a role while `add_role` adds another one. It's a good practice to show how to identify the roles in the recipe, even if it isn't used, for reference. But, we could also select variables automatically using the added role. For example, apply a transformation step to all predictors and select or exclude `has_role("group")`.

Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: I have never worked with recipes myself and don't know about recommended specifications. Feel free to change it as you see fit.

File: vignettes/GetStarted.Rmd (hunk 3 above)
Comment (paul-buerkner) [reply vignettes/GetStarted.Rmd]: Ok. changed it accordingly.

**Comment by hsbadr** [comment]
Now, it fails with:
```r
Error: processing vignette 'GetStarted.Rmd' failed with diagnostics:
argument "RMU" is missing, with no default
```

**Comment by paul-buerkner** [comment]
I think it is a problem with rstan when trying to build the vignette. I changed the header to only eval when not on CRAN. When building vignettes, you have to use 

```R
withr::with_envvar(c("NOT_CRAN" = "true"),
  devtools::build_vignettes()
)
```

to make sure it evalutes. This is the workflow I use in brms to avoid anyone accidentally running models when building the package.

**Comment by hsbadr** [comment]
It still fails with the same error:

> ```r
> Error: processing vignette 'GetStarted.Rmd' failed with diagnostics:
> argument "RMU" is missing, with no default
> ```

**Comment by paul-buerkner** [comment]
I am sorry, I don't know where this comes from as it doesn't fail for me
locally.

Hamada S. Badr ***@***.***> schrieb am So., 16. Mai 2021,
21:45:

**Comment by hsbadr** [comment]
> I am sorry, I don't know where this comes from as it doesn't fail for me locally.

Which version of `withr` do you have installed?

**Comment by paul-buerkner** [comment]
Version 2.4.2

Am So., 16. Mai 2021 um 21:56 Uhr schrieb Hamada S. Badr <
***@***.***>:

**Comment by hsbadr** [comment]
I see. I define `NOT_CRAN` in the GHA workflow.
https://github.com/hsbadr/bayesian/blob/5572a32c224efbe91bd9b2d11ab81ff69dc49cc2/.github/workflows/R.yml#L113

**Comment by hsbadr** [comment]
LGTM. Thanks @paul-buerkner!

**Pull request merged**